### PR TITLE
Upgrade toolchain to 08/28

### PR DESCRIPTION
**Pull request by jaisnan**
Upgrades toolchain to 08/28

Culprit upstream changes:
1. https://github.com/rust-lang/rust/pull/128812
2. https://github.com/rust-lang/rust/pull/128703
3. https://github.com/rust-lang/rust/pull/127679
4. https://github.com/rust-lang/rust-clippy/pull/12993
5. https://github.com/rust-lang/cargo/pull/14370
6. https://github.com/rust-lang/rust/pull/128806

Resolves https://github.com/model-checking/kani/issues/3429
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: tools/scanner/Cargo.toml [hunk 1]
```diff
@@ -20,4 +20,3 @@ strum_macros = "0.26"
 # This crate uses rustc crates.
 # More info: https://github.com/rust-analyzer/rust-analyzer/pull/7891
 rustc_private = true
-
```
Comment (tautschnig) [inline tools/scanner/Cargo.toml:23]: Why is this change needed?

File: kani-compiler/src/kani_middle/points_to/points_to_analysis.rs [hunk 2]
```diff
@@ -53,7 +53,7 @@ struct PointsToAnalysis<'a, 'tcx> {
     tcx: TyCtxt<'tcx>,
     /// This will be used in the future to resolve function pointer and vtable calls. Currently, we
     /// can resolve call graph edges just by looking at the terminators and erroring if we can't
-    /// resolve the callee.  
+    /// resolve the callee.
```
Comment (tautschnig) [inline kani-compiler/src/kani_middle/points_to/points_to_analysis.rs:56]: Whitespace change?

**Review by tautschnig — APPROVED** [review]
Not a big deal, but I'd appreciate links to the upstream changes that forced us to make changes. Might be a bit hard for such a large data range, though.

File: kani-compiler/src/kani_middle/points_to/points_to_analysis.rs (hunk 2 above)
Comment (jaisnan) [reply kani-compiler/src/kani_middle/points_to/points_to_analysis.rs:56]: This is being enforced by cargo fmt, not a manual change.

File: kani-compiler/src/kani_middle/points_to/points_to_analysis.rs [hunk 3]
```diff
@@ -521,7 +521,7 @@ impl<'a, 'tcx> PointsToAnalysis<'a, 'tcx> {
             | Rvalue::ShallowInitBox(operand, _)
             | Rvalue::Cast(_, operand, _)
             | Rvalue::Repeat(operand, ..) => self.successors_for_operand(state, operand),
-            Rvalue::Ref(_, _, ref_place) | Rvalue::AddressOf(_, ref_place) => {
+            Rvalue::Ref(_, _, ref_place) | Rvalue::RawPtr(_, ref_place) => {
```
Comment (carolynzech) [inline kani-compiler/src/kani_middle/points_to/points_to_analysis.rs:524]: Please add this PR: https://github.com/rust-lang/rust/pull/127679
to the issue description of upstream changes that make this toolchain update necessary.

Comment (jaisnan) [reply kani-compiler/src/kani_middle/points_to/points_to_analysis.rs:524]: Done! Thanks for finding that

File: tools/scanner/src/bin/scan.rs [hunk 4]
```diff
@@ -14,6 +14,7 @@
 //! together with the name of the analysis.
 //!
 //! Look at each analysis documentation to see which files an analysis produces.
+#![feature(rustc_private)]
```
Comment (carolynzech) [inline tools/scanner/src/bin/scan.rs:17]: Why do we need this?

File: kani-compiler/src/session.rs [hunk 5]
```diff
@@ -61,7 +61,8 @@ static JSON_PANIC_HOOK: LazyLock<Box<dyn Fn(&panic::PanicHookInfo<'_>) + Sync +
                 Lrc::new(SourceMap::new(FilePathMapping::empty())),
                 fallback_bundle,
                 false,
-                HumanReadableErrorType::Default(ColorConfig::Never),
+                HumanReadableErrorType::Default,
```
Comment (carolynzech) [inline kani-compiler/src/session.rs:64]: Also add https://github.com/rust-lang/rust/pull/128806

**Review by carolynzech — CHANGES_REQUESTED** [review]

File: kani-compiler/src/session.rs (hunk 5 above)
Comment (jaisnan) [reply kani-compiler/src/session.rs:64]: Was already added, its item 6 in the list of PR's

File: tools/scanner/src/bin/scan.rs (hunk 4 above)
Comment (jaisnan) [reply tools/scanner/src/bin/scan.rs:17]: Good question! this [commit](736a249954aedaff9b3562f073e295d6c4c6d94d) essentially mandates this change being required

 Without this change, 

we now get the following error:

```
error: cannot satisfy dependencies so `rustc_driver` only shows up once
  |
  = help: having upstream crates all available in one format will likely make this go away
  = note: `rustc_driver` was unavailable as a static crate, preventing fully static linking
  = help: `feature(rustc_private)` is needed to link to the compiler's `rustc_driver` library

error: cannot satisfy dependencies so `core` only shows up once
  |
  = help: having upstream crates all available in one format will likely make this go away
  = help: `feature(rustc_private)` is needed to link to the compiler's `rustc_driver` library
```

which shows that rustc_driver cannot be linked without adding this feature.

Comment (carolynzech) [reply tools/scanner/src/bin/scan.rs:17]: Link is broken but sounds good :)

**Review by carolynzech — APPROVED** [review]

File: tools/scanner/src/bin/scan.rs (hunk 4 above)
Comment (jaisnan) [reply tools/scanner/src/bin/scan.rs:17]: Fixed link, thanks
